### PR TITLE
Inati

### LINKED ISSUE #64: Csm inati method
could be implemented via a wrapper for https://github.com/ismrmrd/ismrmrd-python-tools/blob/17fe6fbf1bb645112e2ad0023eb4f42afb36421e/ismrmrdtools/coils.py#L92

### PR DESCRIPTION
**Pull request by JoHa0811**
Implements the Inati CSM method and integrates the new folder style 
closes #64 and #161

### CODE REVIEW

File: src/mrpro/data/_CsmData/_CsmData.py [hunk 1]
```diff
@@ -93,3 +93,21 @@ def from_idata_walsh(
         csm_data = csm_fun(idata.data)
 
         return cls(header=idata.header, data=csm_data)
+
+    @classmethod
+    def from_idata_inati(cls, data: torch.Tensor, ks: int, power: int, padding_mode='circular') -> CsmData:
```
Comment (schuenke) [inline src/mrpro/data/_CsmData/_CsmData.py]: if the method is called `from_idata...` I would expect that data is of type IData

File: src/mrpro/data/_CsmData/_CsmData.py [hunk 2]
```diff
@@ -93,3 +93,21 @@ def from_idata_walsh(
...
+        """Coil sensitivity maps using the Inati method.
+
+        Details of the method can be found in Inati et al. 2004.
+
+        Parameters
+        ----------
+        data: Images of shape (coil, E1, E0)
+        ks: kernel size
+        power: number of iterations
+        padding_mode: padding mode for the sliding window
```
Comment (schuenke) [inline src/mrpro/data/_CsmData/_CsmData.py:108]: would be nice to use the same style as in all other methods...

```python
"""
...
Parameters
----------
data
    images of shape...
ks
   kernel size
...
"""

```

File: src/mrpro/data/_CsmData/_inati.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+def coil_map_study_2d_Inati(data: torch.Tensor, ks: int, power: int, padding_mode='circular'):
+    """Coil sensitivity maps using the method described in Inati et al. 2004.
+
+    Parameters
+    ----------
+    data: Images of shape (coil, E1, E0)
+    ks: kernel size
+    power: number of iterations
+    padding_mode: padding mode for the sliding window
```
Comment (schuenke) [inline src/mrpro/data/_CsmData/_inati.py:14]: see comment about style from above

File: src/mrpro/data/_CsmData/_inati.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import torch
+
+from mrpro.utils.window import sliding_window
+
+
+def coil_map_study_2d_Inati(data: torch.Tensor, ks: int, power: int, padding_mode='circular'):
+    """Coil sensitivity maps using the method described in Inati et al. 2004.
```
Comment (schuenke) [inline src/mrpro/data/_CsmData/_inati.py:7]: Maybe use a `References` section as recommended in the [official style guide](https://numpydoc.readthedocs.io/en/latest/format.html#references)

or at least list the complete information including journal, doi etc

Comment (JoHa0811) [reply src/mrpro/data/_CsmData/_inati.py:7]: Yes, good call!
Only problem is that i don't know which paper this is from, as this was the only reference in @fzimmermann89 implementation. I will happily beautify this, if someone could give me a link to the paper :)

Comment (ckolbPTB) [reply src/mrpro/data/_CsmData/_inati.py:7]: 1. Inati S J, Hansen, M. S. & Peter, K. A solution to the phase problem in adaptvie coil combination. in Proceedings of the 21st Annual Meeting of ISMRM, Salt Lake City, USA, 2672 (2013).
2. Inati, S. & Hansen, M. A Fast Optimal Method for Coil Sensitivity Estimation and Adaptive Coil Combination for Complex Images. in Proceedings of Joint Annual Meeting ISMRM-ESMRMB, Milan, Italy, 7115 (2014).

**Comment by fzimmermann89** [comment]
The biggest issues with this code are that we need to 
a) check if it reproduces ptbpyrecon
b) extend it to 3d
c) proper tests
I am not to convinced by having an example for this.

As this is still just a draft that only copied over my code, I would abstain from any further comments and wait until we have #161 merged (this was on purpose a separate PR first)

I do not agree that this suppresseds #161 as all issues mypy complained about were silenced here by removing the typehints.

Please don't do that!

Instead, you are invited to continue working on #161 first.

**Comment by ckolbPTB** [comment]
Superseded by #345

**Pull request closed without merging**